invenio_stats/aggregations.py

```diff
@@ -345,11 +374,18 @@ def agg_iter(self, dt):
                 interval_date = datetime.strptime(
                     doc["timestamp"], "%Y-%m-%dT%H:%M:%S"
                 ).replace(**dict.fromkeys(INTERVAL_ROUNDING[self.interval], 0))
-
+                if aggregation["last_update"]["value"] and previous_bookmark:
+                    # Let's make sure that both objects are of the same type. This is important for the test
+                    previous = float(previous_bookmark.strftime("%s"))
```

Comment:
we can compute this outside the for loop, to avoid redoing `strftime` each time.